File: gym/static/js/facility.js
$(()=>{
    $("#insert").click(()=>{
      if($("#fid").val()==""&&$("#status_label").val()==""&&$("#manufactory").val()==""&&$("#manufacture_date").val()=="" && $("#eid").val()=="" && $("#is_outdoor").val()==""){
          alert("please input content !")
      }
      else {
          $.post(
          '/facility_insert',
          {
              fid:$("#fid").val(),
              status_label:$("#status_label").val(),
              manufactory:$("#manufactory").val(),
              manufacture_date:$("#manufacture_date").val(),
              eid:$("#eid").val(),
              is_outdoor:$("#is_outdoor").val()
          },
          (json)=>{
               if(json.status==1)
                   alert("insert success!")
              if(json.status==0)
                  alert("insert fail please input correct eid and fid !")
          }
      )
      }

    });

    $("#select").click(()=>{
        $("#table").empty();
        $.post(
            '/facility_select',
            {
                fid:$("#fid").val(),
                is_outdoor: $("#is_outdoor").val()
            },
            (json)=>{
                th="<tr>\n" +
                       "<th>Facility ID</th>\n" +
                       "<th>Status label</th>\n" +
                       "<th>Manufactory</th>\n" +
                       "<th>Manufacture_date</th>\n" +
                       "<th>Employee ID</th>\n" +
                       "<th>Is outdoor</th>\n" +
                       "</tr>";
                   $("#table").append(th);
                   json.forEach(
                       (row)=>{
                            $("#table").append(
                                "<tr><td>"+row.fid+"</td><td>"+
                                row.status_label+"</td><td>"+row.manufactory+"</td><td>"+
                                row.manufacture_date+"</td><td>"+row.eid+"</td><td>"+
                                row.is_outdoor+"</td></tr>"
                            )
                       }
                   );

            }

        )
    });



    $("#delete").click(
        ()=>{
            if($("#fid").val()==""){
                alert("please input fid !")
            }
            else {
                $.getJSON(
                '/facility_delete?fid='+$("#fid").val(),
                (json)=>{
                    if(json.status==1)
                        alert("delete success!")
                    else if(json.status==0)
                        alert("delete fail ! please try again")
                    else
                        alert("unknown error, please try again !")
                }
            )
            }
        }
    );

    $("#update").click(()=>{
      if($("#fid").val()==""&&$("#status_label").val()==""&&$("#manufactory").val()==""&&$("#manufacture_date").val()==""&&$("#eid").val()==""&&$("#is_outdoor").val()==""){
          alert("please input content !");
      }
      else {
          $.post(
          '/facility_update',
          {
              fid:$("#fid").val(),
              status_label:$("#status_label").val(),
              manufactory:$("#manufactory").val(),
              manufacture_date:$("#manufacture_date").val(),
              eid:$("#eid").val(),
              is_outdoor:$("#is_outdoor").val()
          },
          (json)=>{
               if(json.status==1)
                   alert("update success!")
              if(json.status==0)
                  alert("update fail please input correct eid and fid !")
          }
      )
      }
    });
})

File: gym/views/facility.py
from gym import app
from flask import render_template, request, jsonify
from gym.daos.dao_facility import DaoFacility
from gym.daos.dao_employee import DaoEmployee


@app.route('/facility_select',methods=['POST'])
def facility_select():
    fid = request.form.get('fid')
    is_outdoor=request.form.get('is_outdoor')
    dao_facility = DaoFacility()
    if fid == '' and is_outdoor == '':
        all_rows = dao_facility.find_all()
    elif fid!='' and is_outdoor != '':
        all_rows = dao_facility.find_facility(int(fid),int(is_outdoor))
    elif fid!='' and is_outdoor == '':
        all_rows=dao_facility.find_fid(int(fid))
    elif fid=='' and int(is_outdoor)==1:
        all_rows=dao_facility.find_outdoor()
    elif (fid is  None or fid=='') and (int(is_outdoor)==0):
        all_rows = dao_facility.find_indoor()
    return jsonify(all_rows)


@app.route('/facility_insert',methods=['POST'])
def facility_insert():
    fid=int(request.form.get('fid'))
    status_label=request.form.get('status_label')
    manufactory=request.form.get('manufactory')
    manufacture_date=request.form.get('manufacture_date')
    eid=int(request.form.get('eid'))
    is_outdoor=int(request.form.get('is_outdoor'))
    dao_facility=DaoFacility()
    dao_employee=DaoEmployee()
    employee_rows=dao_employee.find_all()
    facility_rows=dao_facility.find_all()
    eid_list=list()
    fid_list=list()
    for e in employee_rows:
        eid_list.append(int(e['eid']))
    for f in facility_rows:
        fid_list.append(int(f['fid']))
    if all((fid not in fid_list, eid in eid_list)):
        dao_facility.insert(fid, status_label, manufactory, manufacture_date, eid, is_outdoor)
        return jsonify(status=1)
    else:
        return jsonify(status=0)



@app.route('/facility_delete',methods=['GET'])  #$.getJson is delivered in get way
def facility_delete():
    fid=int(request.args.get('fid'))
    dao_facility = DaoFacility()
    facility_rows = dao_facility.find_all()
    fid_list = list()
    for f in facility_rows:
        fid_list.append(int(f['fid']))
    if fid in fid_list:
        dao_facility.delete(fid)
        return jsonify(status=1)
    else:
        return jsonify(status=0)



@app.route('/facility_update',methods=['POST'])
def facility_update():
    fid = int(request.form.get('fid'))
    status_label = request.form.get('status_label')
    manufactory = request.form.get('manufactory')
    manufacture_date = request.form.get('manufacture_date')
    eid = int(request.form.get('eid'))
    is_outdoor = int(request.form.get('is_outdoor'))
    dao_facility = DaoFacility()
    dao_employee = DaoEmployee()
    employee_rows = dao_employee.find_all()
    facility_rows = dao_facility.find_all()
    eid_list = list()
    fid_list = list()
    for e in employee_rows:
        eid_list.append(int(e['eid']))
    for f in facility_rows:
        fid_list.append(int(f['fid']))
    if fid in fid_list and eid in eid_list:
        dao_facility.update(status_label, manufactory, manufacture_date, eid, is_outdoor,fid)
        return jsonify(status=1)
    else:
        return jsonify(status=0)

